hikari/impl/entity_factory.py

```diff
@@ -987,6 +987,9 @@ def deserialize_member(
 
         role_ids = [snowflakes.Snowflake(role_id) for role_id in payload["roles"]]
 
+        if guild_id not in role_ids:
```

Comment:
Is it possible for them to send it?

Reply:
they could decide to start including it someday without warning, that's just protection against weird behaviour

Reply:
:+1:

Reply:
why aren't we using an ID table here?

Reply:
Is order here meaningful? Could `set` be used here instead of `list`? No duplicate checks would be necessary then.

Reply:
yeah, I agree with @tomxey on this. If we used IDTable here, that implements the set interface, but is implemented as a snowflake array internally as a bisect binary tree.